app/assets/javascripts/select-kit/addon/components/category-selector.js

```diff
@@ -52,6 +54,57 @@ export default MultiSelectComponent.extend({
     return "category-row";
   },
 
+  search(filter) {
+    const result = this._super(filter);
+    if (result.length === 1) {
```

Comment:
Does this work when there are 3 levels?

Reply:
Yes it does. The `1` here represents that the "+ all subcategories" option shows only if it would normally show only one category (i.e. perfect match). I did not want to double each option with a "+ all subcategories" option.